src/test/ui/consts/const-eval/const_panic_2021.rs

```diff
@@ -0,0 +1,27 @@
+// edition:2021
+#![feature(const_panic)]
+#![crate_type = "lib"]
+
+const A: () = std::panic!("blåhaj");
```

Comment:
So... if my understanding is right, `std::panic!(SOME_CONST)` does not work anymore in 2021, does your change permit `panic!("{}", SOME_CONST_STR)` to work? If so, that should have a test.

Reply:
oof, our test suite is really subpar. We have zero tests for

```rust
const FOO: () = {
    let mut x = [4, 2];
    for i in 0..x.len() {
         x[i] = x[i] + b'0';
    }
    let z = unsafe { std::str::from_utf8_unchecked(&x) };
    panic!("{}", z);
};
```

Reply:
> So... if my understanding is right, `std::panic!(SOME_CONST)` does not work anymore in 2021, does your change permit `panic!("{}", some_str)` to work?

Nope. I want to make a next PR that makes `panic!("{}", "literal")` work. `panic!("{}", some_str_var)` will be a step further into the future.